app/src/main/java/com/example/tabatatimer/SettingsActivity.kt

```diff
@@ -0,0 +1,58 @@
+package com.example.tabatatimer
+
+import android.content.SharedPreferences
+import android.os.Bundle
+import androidx.appcompat.app.AppCompatActivity
+import androidx.appcompat.app.AppCompatDelegate
+import androidx.preference.ListPreference
+import androidx.preference.Preference
+import androidx.preference.PreferenceFragmentCompat
+import androidx.preference.PreferenceManager
+
+class SettingsActivity : AppCompatActivity(), SharedPreferences.OnSharedPreferenceChangeListener,
+    Preference.SummaryProvider<ListPreference> {
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.fragment_settings)
+        supportFragmentManager
+            .beginTransaction()
+            .replace(R.id.settings, SettingsFragment())
+            .commit()
+        supportActionBar?.setDisplayHomeAsUpEnabled(true)
+
+        PreferenceManager.getDefaultSharedPreferences(this)
+            .registerOnSharedPreferenceChangeListener(this)
+    }
+
+    override fun onSharedPreferenceChanged(sharedPreferences: SharedPreferences?, key: String?) {
+        val darkModeString = getString(R.string.dark_mode)
+        key?.let {
+            if (it == darkModeString) sharedPreferences?.let { pref ->
+                val darkModeValues = resources.getStringArray(R.array.dark_mode_values)
+                when (pref.getString(darkModeString, darkModeValues[0])) {
+                    darkModeValues[0] -> AppCompatDelegate.setDefaultNightMode(AppCompatDelegate.MODE_NIGHT_FOLLOW_SYSTEM)
+                    darkModeValues[1] -> AppCompatDelegate.setDefaultNightMode(AppCompatDelegate.MODE_NIGHT_NO)
+                    darkModeValues[2] -> AppCompatDelegate.setDefaultNightMode(AppCompatDelegate.MODE_NIGHT_YES)
+                    darkModeValues[3] -> AppCompatDelegate.setDefaultNightMode(AppCompatDelegate.MODE_NIGHT_AUTO_BATTERY)
+                }
+            }
+        }
```

Comment:
I wasn't able to detect the difference between light and dark modes.

Reply:
I fixed it, the reason was wrong starting SettingActivity. You can check the commit below.
![photo_2020-12-30_19-41-58 (2)](https://user-images.githubusercontent.com/41323462/103367917-4f12fa80-4ad7-11eb-800d-bcd7228f6b1a.jpg)
![photo_2020-12-30_19-41-58 (3)](https://user-images.githubusercontent.com/41323462/103367919-4fab9100-4ad7-11eb-9c01-5e474b72d710.jpg)
![photo_2020-12-30_19-41-58](https://user-images.githubusercontent.com/41323462/103367920-4fab9100-4ad7-11eb-84a8-ba7284eac857.jpg)